internal/controller/eventing/utils_test.go

```diff
@@ -0,0 +1,93 @@
+package eventing
+
+import (
+	"context"
+	"testing"
+
+	eventingv1alpha1 "github.com/kyma-project/eventing-manager/api/v1alpha1"
+	"github.com/kyma-project/eventing-manager/test/utils"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_containsFinalizer(t *testing.T) {
+	t.Parallel()
+
+	// define test cases
+	testCases := []struct {
+		name          string
+		givenEventing *eventingv1alpha1.Eventing
+		wantResult    bool
+	}{
+		{
+			name:          "should return false when finalizer is missing",
+			givenEventing: utils.NewEventingCR(),
+			wantResult:    false,
+		},
+		{
+			name:          "should return true when finalizer is present",
+			givenEventing: utils.NewEventingCR(utils.WithEventingCRFinalizer(FinalizerName)),
+			wantResult:    true,
+		},
+	}
+
+	// run test cases
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			// given
+			testEnv := NewMockedUnitTestEnvironment(t)
+			reconciler := testEnv.Reconciler
+
+			// when, then
+			require.Equal(t, tc.wantResult, reconciler.containsFinalizer(tc.givenEventing))
+		})
+	}
+}
+
+func Test_addFinalizer(t *testing.T) {
+	t.Parallel()
+
+	t.Run("should add finalizer", func(t *testing.T) {
+		t.Parallel()
```

Comment:
```suggestion
```
since you already set t.Parallel() one level above and since you don't run a table test, this t.Parallel() is redundant.